insights/parsers/grubenv.py

```diff
@@ -90,3 +74,74 @@ def has_tuned_params(self):
     @property
     def tuned_params(self):
         return self.get("tuned_params", "")
+
+
+@parser(Specs.grubenv)
+class GrubEnv(GrubenvBase):
+    """
+    Parses the ``/boot/grub2/grubenv`` file and returns a dict of the grubenv variables.
+
+    Sample output of the file::
+
+        saved_entry=295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64
+        kernelopts=root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200
+        boot_success=0
+        boot_indeterminate=2
+        tuned_params=transparent_hugepages=never
+        tuned_initrd=
+
+    Examples:
+        >>> type(grubenv)
+        <class 'insights.parsers.grubenv.GrubEnv'>
+        >>> grubenv.has_kernelopts
+        True
+        >>> grubenv.kernelopts
+        'root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200'
+        >>> grubenv.has_tuned_params
+        True
+        >>> grubenv.tuned_params
+        'transparent_hugepages=never'
+        >>> grubenv['saved_entry']
+        '295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64'
+    """
+    pass
+
+
+@parser(Specs.grubenv)
+class Grub2EditenvList(GrubenvBase):
+    """
+    This parser parses the output of the command ``grub2-editenv list``, which
+    list the current variables in GRUB environment block file or error message.
+
+    The parser processes the ``grub2-editenv list`` command and returns a dict
+    of the grubenv variables. The command output without error message is same
+    as the content in ``/boot/grub2/grubenv``.
+
+    Sample output of the command::
+
+        saved_entry=295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64
+        kernelopts=root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200
+        boot_success=0
+        boot_indeterminate=2
+        tuned_params=transparent_hugepages=never
+        tuned_initrd=
+
+    Examples:
+
+        >>> type(grub2_editenv_list)
+        <class 'insights.parsers.grubenv.Grub2EditenvList'>
+        >>> grub2_editenv_list.has_kernelopts
+        True
+        >>> grub2_editenv_list.kernelopts
+        'root=/dev/mapper/root_vg-lv_root ro crashkernel=auto resume=/dev/mapper/root_vg-lv_swap rd.lvm.lv=root_vg/lv_root rd.lvm.lv=root_vg/lv_swap console=tty0 console=ttyS0,115200'
+        >>> grub2_editenv_list.has_tuned_params
+        True
+        >>> grub2_editenv_list.tuned_params
+        'transparent_hugepages=never'
+        >>> grub2_editenv_list['saved_entry']
+        '295e1ba1696e4fad9e062f096f92d147-4.18.0-305.el8.x86_64'
+    """
+
+    @property
+    def error(self):
```

Comment:
Could this be moved to the base class so both classes can use the property? Also could this be changed to errors since it's returning a list, and add an entry to the attributes docstring in the base class.

Reply:
Like the comment I left above, the content of 'GrubEnv' parser will never contain error messages. So, the error attribute is no need to be added in base class for each parser to use.